catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/SettingsActivityCrashReporterTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.ui.activity;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+
+import org.catrobat.catroid.ui.SettingsActivity;
+import org.catrobat.catroid.utils.CrashReporter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+public class SettingsActivityCrashReporterTest {
+
+	@Before
+	public void setUp() throws Exception {
+		CrashReporter.setIsCrashReportEnabled(true);
+	}
+
+	@Test
+	public void testAutoCrashReportEnabling() {
+		Context context = InstrumentationRegistry.getTargetContext();
+
+		SettingsActivity.setAutoCrashReportingEnabled(context, false);
+
+		assertFalse(CrashReporter.isInitialized);
+
+		SettingsActivity.setAutoCrashReportingEnabled(context, true);
+
+		assertTrue(CrashReporter.isInitialized);
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		CrashReporter.isInitialized = false;
```

Comment:
In this case, the setUp and tearDown should do the same thing (call the same methods) I guess.